### ISSUE TITLE
Read-Only databases should still allow calling `dolt_checkout()`

### ISSUE THREAD

**Issue by fulghum** [dolthub/dolt#6730]
In Dolt version 1.16.4, if a sql-server is running in read-only mode, you are no longer able to call `dolt_checkout()` from within a session.

```sql
mysql> CALL DOLT_CHECKOUT('mainline');
ERROR 1105 (HY000): database server is set to read only mode
``` 

Additionally, `dolt status` from the command line returns the same read only mode error:

```bash
❯ dolt status
Error 1105 (HY000): database server is set to read only mode
```

**Comment by fulghum** [comment]
The customer also reported that calling `dolt status` from the command line, while a sql-server is running from that direction in read-only mode, results in the same error, but I haven't been able to repro that case yet.

**Comment by fulghum** [comment]
The fix for this issue has been released in [Dolt version 1.16.5](https://github.com/dolthub/dolt/releases/tag/v1.16.5).
